src/components/AutoComplete/AutoComplete.js

```diff
@@ -24,165 +24,100 @@ function renderMenu(items, value, style) {
   return <div className="ac-menu" children={items} />;
 }
 
-export default class AutoComplete extends Component {
-  state = {
-    value: '',
+export default function AutoComplete(props) {
+  const { showClear, source, 'data-testid': testid, id } = props;
+  const [value, setValue] = useState('');
+
+  const setPermalinkParams = (param, value) => {
+    const permalink = new URLSearchParams(window.location.search);
+    const paramItem = source.filter(item => item.label === value);
+    if (paramItem.length && value.length) {
+      permalink.set(param, paramItem[0].value);
+    } else if (!value.length) {
+      permalink.delete(param);
+    }
+    window.history.pushState({}, null, '?' + permalink.toString());
   };
 
-  onSelect = (value, item) => {
-    this.setState({ value });
-    this.props.onSelect(item);
-    this.setPermalinkParams(this.props.id, value);
+  const getPermalinkParams = () => {
+    const permalink = new URLSearchParams(window.location.search);
+    const paramValue = permalink.get(props.id);
+    const paramItem = source.filter(item => item.value === paramValue);
+    if (paramItem.length) {
+      setValue(paramItem[0].label);
+      props.onSelect({ value: paramValue, label: paramItem[0].label });
+    }
   };
 
-  onChange = (event, value) => {
-    this.setState({ value });
+  useEffect(getPermalinkParams, [source.length]);
```

Comment:
I'm not about this `source.length` could lead to a minor bug where the list changes but not the length, to rely on `source` we would need a memoized source from the parent component or the effect will run on every rerender.

Reply:
Do you have a better approach?